backend/src/apiserver/model/pipeline_version.go

```diff
@@ -0,0 +1,65 @@
+package model
+
+import "fmt"
+
+// PipelineVersionStatus a label for the status of the Pipeline.
+// This is intend to make pipeline creation and deletion atomic.
+type PipelineVersionStatus string
+
+const (
+	PipelineVersionCreating PipelineVersionStatus = "CREATING"
+	PipelineVersionReady    PipelineVersionStatus = "READY"
+	PipelineVersionDeleting PipelineVersionStatus = "DELETING"
+)
+
+type PipelineVersion struct {
+	UUID           string `gorm:"column:VersionUUID; not null; primary_key"`
+	CreatedAtInSec int64  `gorm:"column:VersionCreatedAtInSec; not null"`
+	Name           string `gorm:"column:VersionName; not null"`
+	// Set size to 65535 so it will be stored as longtext.
+	// https://dev.mysql.com/doc/refman/8.0/en/column-count-limit.html
+	Parameters string                `gorm:"column:VersionParameters; not null; size:65535"`
+	PipelineId string                `gorm:"column:PipelineId; not null"`
+	Status     PipelineVersionStatus `gorm:"column:VersionStatus; not null"`
+	CodeSource
+}
+
+type CodeSource struct {
+	// PipelineVersion can be based on a git commit or a package stored at some
+	// url.
+	// All fields below are optional.
```

Comment:
for storing oneof fields in DB should we instead use just two columns? code source type and the streamed json of the codesource?
This will make DB schema much cleaner and extensible.

Reply:
I get rid of the oneof from api type definition. So kept 3 cols here for now. Please let me if we still want 2 columns instead: one is the combination of repo name and commit; the other is url.